install_config/upgrading/known_issues.adoc

```diff
@@ -0,0 +1,227 @@
++
+----
+$ sh rolebindingrestrictions.sh
+----
+
+[[upgrading-known-issue-BZ1482978]]
+== Orphaned OAuthClientAuthorization Objects
+
+Orphaned `OAuthClientAuthorization` objects can occur during the upgrade due to
+the deletion of the associated OAuth client.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1482978[*BZ#1482978*])
+
+.Error Output
+----
+TASK [Upgrade all storage] *********************************************************************************************
+task path: /usr/share/ansible/openshift-ansible/playbooks/common/openshift-cluster/upgrades/upgrade_control_plane.yml:11
+Using module file /usr/lib/python2.7/site-packages/ansible/modules/commands/command.py
+
+fatal: [test.linux.lan]: FAILED! => {
+    "changed": true,
+    "cmd": [
+        "oc",
+        "adm",
+        "--config=/etc/origin/master/admin.kubeconfig",
+        "migrate",
+        "storage",
+        "--include=*",
+        "--confirm"
+    ],
+    "delta": "0:00:56.843544",
+    "end": "2017-08-17 12:46:36.417413",
+    "failed": true,
+    "failed_when_result": true,
+    "invocation": {
+        "module_args": {
+            "_raw_params": "oc adm --config=/etc/origin/master/admin.kubeconfig migrate storage --include=* --confirm",
+            "_uses_shell": false,
+            "chdir": null,
+            "creates": null,
+            "executable": null,
+            "removes": null,
+            "warn": true
+        }
+    },
+    "rc": 1,
+    "start": "2017-08-17 12:45:39.573869"
+}
+
+STDOUT:
+
+
+error: <1>
+oauthclientauthorizations/user1:system:serviceaccount:namespace1:jenkins : OAuthClientAuthorization "user1:system:serviceaccount:namespace1:jenkins" is invalid: clientName: Internal error: system:serviceaccount:namespace1:jenkins has no redirectURIs; set serviceaccounts.openshift.io/oauth-redirecturi.<some-value>=<redirect> or create a dynamic URI using serviceaccounts.openshift.io/oauth-redirectreference.<some-value>=<reference>
+
+error: <2>
+oauthclientauthorizations/user2:system:serviceaccount:namespace2:jenkins : OAuthClientAuthorization "user2:system:serviceaccount:namespace2:jenkins" is invalid: clientName: Internal error: serviceaccounts "jenkins" not found
+summary: total=1002 errors=2 ignored=0 unchanged=1000 migrated=0
+info: to rerun only failing resources, add --include=oauthclientauthorizations
+error: 2 resources failed to migrate
+----
+<1> This error occurs when the service account based OAuth client has no valid redirect URIs.
+<2> Reason
+
+The first error can be solved in one of two ways:
+
+- Annotate the service account with a valid redirect URI; see xref:../../architecture/additional_concepts/authentication.adoc#redirect-uris-for-service-accounts[Redirect URIs for Service Accounts as OAuth Clients].
+
+Or:
+
+- Delete the service account in the offending namespace if it is not being used
+anymore:
++
+----
+$ oc delete serviceaccount jenkins -n <namespace>
```

Comment:
Change jenkins to `<saname>` since we are generalizing it?

Reply:
:+1: